src/components/MDXComponents/__tests__/MDXLink.test.tsx

```diff
@@ -0,0 +1,155 @@
+import { render, screen } from '@testing-library/react';
+import { MDXLink } from '../MDXLink';
+
+const routerMock = {
+  __esModule: true,
+  useRouter: () => {
+    return {
+      query: {
+        platform: ''
+      },
+      asPath: ''
+    };
+  }
+};
+
+jest.mock('next/router', () => routerMock);
+
+jest.mock('next/link', () => {
+  const MockedNextLink = ({ children, href }) => {
+    const hrefProp =
+      href && typeof href === 'object'
+        ? `${href.pathname.replace('[platform]', href.query?.platform)}${
+            href.hash ? `${href.hash}` : ''
+          }`
+        : href;
+
+    return <a href={hrefProp}>{children}</a>;
+  };
+
+  return MockedNextLink;
+});
```

Comment:
I couldn't get the mock for `next/link` to work without reimplementing it in how I think it should work. This doesn't feel right and so I'm open to any feedback on this.

I tried refactoring out this the logic for setting up the `href` object so I mock `next/link` using `jest.mock('next/link', () => ({ children }) => children);` and tried to spy on it. This didn't work though, kept getting some nextjs errors

Reply:
I tried a bunch of suggestions I found but couldn't get any of them to work either. I'm good with this for now.